epochStart/metachain/legacySystemSCs.go

```diff
@@ -290,7 +290,7 @@ func (s *legacySystemSCProcessor) unStakeNodesWithNotEnoughFunds(
 		}
 
 		validatorLeaving := validatorInfo.ShallowClone()
-		validatorLeaving.SetList(string(common.LeavingList))
+		validatorLeaving.SetList(string(common.LeavingList), s.enableEpochsHandler.IsStakingV4Started())
```

Comment:
you don't need to set the index as well?

Reply:
Yes, very good catch.
Updated index as well